src/Controls/src/Core/Handlers/Items/iOS/ItemsViewController.cs

```diff
@@ -549,15 +575,61 @@ protected virtual CGRect DetermineEmptyViewFrame()
 
 		protected void RemeasureLayout(VisualElement formsElement)
 		{
+			Size size;
 			if (IsHorizontal)
 			{
 				var request = formsElement.Measure(double.PositiveInfinity, CollectionView.Frame.Height);
-				formsElement.Arrange(new Rect(0, 0, request.Width, CollectionView.Frame.Height));
+				size = new Size(request.Width, CollectionView.Frame.Height);
 			}
 			else
 			{
 				var request = formsElement.Measure(CollectionView.Frame.Width, double.PositiveInfinity);
-				formsElement.Arrange(new Rect(0, 0, CollectionView.Frame.Width, request.Height));
+				size = new Size(CollectionView.Frame.Width, request.Height);
+			}
+
+			var platformView = formsElement.ToPlatform();
+			Rect rect;
+			if (platformView.Superview is GeneralWrapperView generalWrapperView)
+			{
+				rect = new Rect(Point.Zero, size);
+				var originalFrame = generalWrapperView.Frame;
+				generalWrapperView.Frame = new CGRect(originalFrame.X, originalFrame.Y, (nfloat)size.Width, (nfloat)size.Height);
+			}
+			else
+			{
+				rect = new Rect(platformView.Frame.X, platformView.Frame.Y, size.Width, size.Height);
+			}
+
+			formsElement.Arrange(rect);
```

Comment:
General question here....

If the parent is a `GeneralWrapperView` is it the right path to call `Arrange` on the `FormsElement`?

or should we call `CrossPlatformArrange` on  `GeneralWrapperView` with the `Frame` you've set on line 596? 

I worry about the `Frame` getting set on like 596 triggering a layout pass that's just going to fire arrange again on  `GeneralWrapperView`